lua/telekasten.lua

```diff
@@ -197,17 +197,23 @@ local monthmap = {
     "December",
 }
 
-local function calenderinfo_today()
-    local dinfo = os.date("*t")
+local dateformats = {
+    date = "%Y-%m-%d",
+    week = "%V",
+    isoweek = "%Y-W%V"
+}
+
+local function calculate_dates(cinfo)
+    local time = os.time(cinfo) -- convert date input to a timestamp 
+    local dinfo = os.date("*t", time) 
+    local oneday = 24 * 60 * 60 -- hours * days * seconds
+    local oneweek = 7 * oneday
+    local oneyear = 365 * oneday
+    local df = dateformats
+
     local opts = {}
-    opts.date = os.date("%Y-%m-%d")
-    local wday = dinfo.wday - 1
-    if wday == 0 then
-        wday = 7
-    end
-    if wday == 6 then
-        wday = 1
-    end
+    opts.date = os.date(df.date)
+    local wday = dinfo.wday - 1 -- compensate for 1-indexed os.date output
```

Comment:
I haven't tested it yet. But in lua, everything is 1-indexed. Could this ever become 0 - and shouldn't it then wrap over to 7?

Reply:
Yeah,  I've been banging my head around this for a while and ended up reverting some of the changes, including the reindexing above.  I had to find the just-right magic combination for the days of week to work for all days, with Sunday considered the last day of the week.

I also ended up finding out that all of lua is 1-indexed as part of this "exercise", so... yay?

Reply:
I saw your push and immediately marked this part as resolved :smile: 

Please test it properly for all sorts of edge cases. The calendar plugin can be a great tool for that. (hint!)

Reply:
@renerocksai  Yup!  I've been using the calendar all afternoon for testing!  I found some issues around the calendar that I want to submit PRs for.  Mostly some documentation around using the calendar filetype to make sure some UI  chrome gets disabled.  My signcolumn, relativenumbers, and alignment markers were causing display issues.

It seems like calendar_monday doesn't get respected either.  No matter what the calendar starts the display on Monday.  Maybe I don't understand that setting.